engine/table/src/main/java/io/deephaven/engine/table/impl/AsOfJoinHelper.java

```diff
@@ -157,8 +168,25 @@ private static Table rightStaticAj(JoinControl control,
             leftGrouping = rightGrouping = null;
         }
 
-        final StaticChunkedAsOfJoinStateManager asOfJoinStateManager =
-                new StaticChunkedAsOfJoinStateManager(leftSources, size, originalLeftSources);
+        final StaticHashedAsOfJoinStateManager asOfJoinStateManager;
+        if (buildLeft) {
+            asOfJoinStateManager = USE_TYPED_STATE_MANAGER
+                    ? TypedHasherFactory.make(StaticAsOfJoinStateManagerTypedBase.class,
+                            leftSources, originalLeftSources,
+                            control.tableSizeForLeftBuild(leftTable),
+                            control.getMaximumLoadFactor(), control.getTargetLoadFactor())
+                    : new StaticChunkedAsOfJoinStateManager(leftSources,
+                            size, originalLeftSources);
+        } else {
+            asOfJoinStateManager = USE_TYPED_STATE_MANAGER
+                    ? TypedHasherFactory.make(StaticAsOfJoinStateManagerTypedBase.class,
+                            leftSources, originalLeftSources,
+                            control.tableSizeForRightBuild(rightTable),
```

Comment:
Probably just set this to a variable,  so there is less manual diffing to do in order to understand the only difference is the size.